src/main/java/org/fsg1/fmms/backend/database/Connection.java

```diff
@@ -67,18 +68,25 @@ public String executeQuery(final String columnName, final String statement, fina
      * @param connection Connection to execute the statement on.
      * @param statement  Statement to perform.
      * @param parameters Array of parameters to map to the statement.
-     * @return A possible result in JSON string form if columnName is given, else an empty string is returned.
+     * @return The generated INSERT id if an INSERT was made, or 0 if no INSERT was made but, for instance,
+     * a DELETE or UPDATE.
      * @throws SQLException If a database access error occurs or anything else goes wrong.
      */
-    public int executeUpdate(final java.sql.Connection connection,
-                             final String statement,
-                             final Object... parameters) throws SQLException {
-        try (PreparedStatement preparedStatement = connection.prepareStatement(statement)) {
+    public long executeUpdate(final java.sql.Connection connection,
+                              final String statement,
+                              final Object... parameters) throws SQLException {
+        try (PreparedStatement preparedStatement = connection.prepareStatement(statement, Statement.RETURN_GENERATED_KEYS)) {
             mapParams(preparedStatement, parameters);
-            return preparedStatement.executeUpdate();
-        } catch (Exception e) {
-            connection.rollback();
-            throw e;
+            preparedStatement.executeUpdate();
+            try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    return generatedKeys.getLong(1);
+                }
+                return 0;
```

Comment:
This should not happen, maybe error handling?

Reply:
It shouldn't? What will it return on a delete, or update?

Reply:
Return -1 and check everytime you depend on this result whether it is real or not